[Quant][core][refactorization] Refactored qconv_unpack.cpp into an implementation file and higher level call registration and definition file

dzdang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #73959
* #73956
* #73857
* #73849
* __->__ #73773
* #73510

Summary:
This refactorization was necessary with the introduction of packed parameters for cudnn.
Specifically, the unpack function for the 3 backends: fbgemm, qnnpack, and cudnn, is called
using dynamic polymorphism, which was previously done in the file (formerly) /quantized/cpu/qconv_unpack.cpp.
This part of the file was moved to the parent directory /quantized/ as it is relevant for both
CPU (fbgemm & qnnpack) & CUDA (cudnn in this case). The remaining content is implementation specific to
CPU, and the file was subsequently renamed to qconv_unpack_impl.cpp

Differential Revision: [D34641680](https://our.internmc.facebook.com/intern/diff/D34641680)

dzdang committed:
[Quant][core][refactorization] Refactored qconv_unpack.cpp into an implementation file and higher level call registration and definition file

Summary:
This refactorization was necessary with the introduction of packed parameters for cudnn.
Specifically, the unpack function for the 3 backends: fbgemm, qnnpack, and cudnn, is called
using dynamic polymorphism, which was previously done in the file (formerly) /quantized/cpu/qconv_unpack.cpp.
This part of the file was moved to the parent directory /quantized/ as it is relevant for both
CPU (fbgemm & qnnpack) & CUDA (cudnn in this case). The remaining content is implementation specific to
CPU, and the file was subsequently renamed to qconv_unpack_impl.cpp

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34641680).

dzdang committed:
Update on "[Quant][core][refactorization] Refactored qconv_unpack.cpp into an implementation file and higher level call registration and definition file"


Summary:
This refactorization was necessary with the introduction of packed parameters for cudnn.
Specifically, the unpack function for the 3 backends: fbgemm, qnnpack, and cudnn, is called
using dynamic polymorphism, which was previously done in the file (formerly) /quantized/cpu/qconv_unpack.cpp.
This part of the file was moved to the parent directory /quantized/ as it is relevant for both
CPU (fbgemm & qnnpack) & CUDA (cudnn in this case). The remaining content is implementation specific to
CPU, and the file was subsequently renamed to qconv_unpack_impl.cpp

Differential Revision: [D34641680](https://our.internmc.facebook.com/intern/diff/D34641680)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34641680).

dzdang committed:
Update on "[Quant][core][refactorization] Refactored qconv_unpack.cpp into an implementation file and higher level call registration and definition file"


Summary:
This refactorization was necessary with the introduction of packed parameters for cudnn.
Specifically, the unpack function for the 3 backends: fbgemm, qnnpack, and cudnn, is called
using dynamic polymorphism, which was previously done in the file (formerly) /quantized/cpu/qconv_unpack.cpp.
This part of the file was moved to the parent directory /quantized/ as it is relevant for both
CPU (fbgemm & qnnpack) & CUDA (cudnn in this case). The remaining content is implementation specific to
CPU, and the file was subsequently renamed to qconv_unpack_impl.cpp

Differential Revision: [D34641680](https://our.internmc.facebook.com/intern/diff/D34641680)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34641680).

dzdang committed:
Update on "[Quant][core][refactorization] Refactored qconv_unpack.cpp into an implementation file and higher level call registration and definition file"


Summary:
This refactorization was necessary with the introduction of packed parameters for cudnn.
Specifically, the unpack function for the 3 backends: fbgemm, qnnpack, and cudnn, is called
using dynamic polymorphism, which was previously done in the file (formerly) /quantized/cpu/qconv_unpack.cpp.
This part of the file was moved to the parent directory /quantized/ as it is relevant for both
CPU (fbgemm & qnnpack) & CUDA (cudnn in this case). The remaining content is implementation specific to
CPU, and the file was subsequently renamed to qconv_unpack_impl.cpp

Differential Revision: [D34641680](https://our.internmc.facebook.com/intern/diff/D34641680)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][refactorization] Refactored qconv_unpack.cpp into an implementation file and higher level call registration and definition file"


Summary:
This refactorization was necessary with the introduction of packed parameters for cudnn.
Specifically, the unpack function for the 3 backends: fbgemm, qnnpack, and cudnn, is called
using dynamic polymorphism, which was previously done in the file (formerly) /quantized/cpu/qconv_unpack.cpp.
This part of the file was moved to the parent directory /quantized/ as it is relevant for both
CPU (fbgemm & qnnpack) & CUDA (cudnn in this case). The remaining content is implementation specific to
CPU, and the file was subsequently renamed to qconv_unpack_impl.cpp

Differential Revision: [D34641680](https://our.internmc.facebook.com/intern/diff/D34641680)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][refactorization] Refactored qconv_unpack.cpp into an implementation file and higher level call registration and definition file"


Summary:
This refactorization was necessary with the introduction of packed parameters for cudnn.
Specifically, the unpack function for the 3 backends: fbgemm, qnnpack, and cudnn, is called
using dynamic polymorphism, which was previously done in the file (formerly) /quantized/cpu/qconv_unpack.cpp.
This part of the file was moved to the parent directory /quantized/ as it is relevant for both
CPU (fbgemm & qnnpack) & CUDA (cudnn in this case). The remaining content is implementation specific to
CPU, and the file was subsequently renamed to qconv_unpack_impl.cpp

Differential Revision: [D34641680](https://our.internmc.facebook.com/intern/diff/D34641680)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][refactorization] Refactored qconv_unpack.cpp into an implementation file and higher level call registration and definition file"


Summary:
This refactorization was necessary with the introduction of packed parameters for cudnn.
Specifically, the unpack function for the 3 backends: fbgemm, qnnpack, and cudnn, is called
using dynamic polymorphism, which was previously done in the file (formerly) /quantized/cpu/qconv_unpack.cpp.
This part of the file was moved to the parent directory /quantized/ as it is relevant for both
CPU (fbgemm & qnnpack) & CUDA (cudnn in this case). The remaining content is implementation specific to
CPU, and the file was subsequently renamed to qconv_unpack_impl.cpp

Differential Revision: [D34641680](https://our.internmc.facebook.com/intern/diff/D34641680)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][refactorization] Refactored qconv_unpack.cpp into an implementation file and higher level call registration and definition file"


Summary:
This refactorization was necessary with the introduction of packed parameters for cudnn.
Specifically, the unpack function for the 3 backends: fbgemm, qnnpack, and cudnn, is called
using dynamic polymorphism, which was previously done in the file (formerly) /quantized/cpu/qconv_unpack.cpp.
This part of the file was moved to the parent directory /quantized/ as it is relevant for both
CPU (fbgemm & qnnpack) & CUDA (cudnn in this case). The remaining content is implementation specific to
CPU, and the file was subsequently renamed to qconv_unpack_impl.cpp

Differential Revision: [D34641680](https://our.internmc.facebook.com/intern/diff/D34641680)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][refactorization] Refactored qconv_unpack.cpp into an implementation file and higher level call registration and definition file"


Summary:
This refactorization was necessary with the introduction of packed parameters for cudnn.
Specifically, the unpack function for the 3 backends: fbgemm, qnnpack, and cudnn, is called
using dynamic polymorphism, which was previously done in the file (formerly) /quantized/cpu/qconv_unpack.cpp.
This part of the file was moved to the parent directory /quantized/ as it is relevant for both
CPU (fbgemm & qnnpack) & CUDA (cudnn in this case). The remaining content is implementation specific to
CPU, and the file was subsequently renamed to qconv_unpack_impl.cpp

Differential Revision: [D34641680](https://our.internmc.facebook.com/intern/diff/D34641680)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][refactorization] Refactored qconv_unpack.cpp into an implementation file and higher level call registration and definition file"


Summary:
This refactorization was necessary with the introduction of packed parameters for cudnn.
Specifically, the unpack function for the 3 backends: fbgemm, qnnpack, and cudnn, is called
using dynamic polymorphism, which was previously done in the file (formerly) /quantized/cpu/qconv_unpack.cpp.
This part of the file was moved to the parent directory /quantized/ as it is relevant for both
CPU (fbgemm & qnnpack) & CUDA (cudnn in this case). The remaining content is implementation specific to
CPU, and the file was subsequently renamed to qconv_unpack_impl.cpp

Differential Revision: [D34641680](https://our.internmc.facebook.com/intern/diff/D34641680)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][refactorization] Refactored qconv_unpack.cpp into an implementation file and higher level call registration and definition file"


Summary:
This refactorization was necessary with the introduction of packed parameters for cudnn.
Specifically, the unpack function for the 3 backends: fbgemm, qnnpack, and cudnn, is called
using dynamic polymorphism, which was previously done in the file (formerly) /quantized/cpu/qconv_unpack.cpp.
This part of the file was moved to the parent directory /quantized/ as it is relevant for both
CPU (fbgemm & qnnpack) & CUDA (cudnn in this case). The remaining content is implementation specific to
CPU, and the file was subsequently renamed to qconv_unpack_impl.cpp

Differential Revision: [D34641680](https://our.internmc.facebook.com/intern/diff/D34641680)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][refactorization] Refactored qconv_unpack.cpp into an implementation file and higher level call registration and definition file"


Summary:
This refactorization was necessary with the introduction of packed parameters for cudnn.
Specifically, the unpack function for the 3 backends: fbgemm, qnnpack, and cudnn, is called
using dynamic polymorphism, which was previously done in the file (formerly) /quantized/cpu/qconv_unpack.cpp.
This part of the file was moved to the parent directory /quantized/ as it is relevant for both
CPU (fbgemm & qnnpack) & CUDA (cudnn in this case). The remaining content is implementation specific to
CPU, and the file was subsequently renamed to qconv_unpack_impl.cpp

Differential Revision: [D34641680](https://our.internmc.facebook.com/intern/diff/D34641680)

[ghstack-poisoned]

jerryzh168 commented on tools/build_variables.bzl:
```diff
@@ -1159,7 +1159,7 @@ aten_native_source_non_codegen_list = [
     "aten/src/ATen/native/quantized/cpu/qconv.cpp",
     "aten/src/ATen/native/quantized/cpu/qconv_prepack.cpp",
-    "aten/src/ATen/native/quantized/cpu/qconv_unpack.cpp",
+    "aten/src/ATen/native/quantized/cpu/qconv_unpack_impl.cpp",
```
do we need to add "aten/src/ATen/native/quantized/qconv_unpack.cpp" to somewhere?

dzdang replied:
I previously had qconv_unpack.cpp added here and it caused those mobile errors in the unit test that we previously discussed. I'm not sure if we need to add this anywhere within this file

dzdang committed:
Update on "[Quant][core][refactorization] Refactored qconv_unpack.cpp into an implementation file and higher level call registration and definition file"


Summary:
This refactorization was necessary with the introduction of packed parameters for cudnn.
Specifically, the unpack function for the 3 backends: fbgemm, qnnpack, and cudnn, is called
using dynamic polymorphism, which was previously done in the file (formerly) /quantized/cpu/qconv_unpack.cpp.
This part of the file was moved to the parent directory /quantized/ as it is relevant for both
CPU (fbgemm & qnnpack) & CUDA (cudnn in this case). The remaining content is implementation specific to
CPU, and the file was subsequently renamed to qconv_unpack_impl.cpp

Differential Revision: [D34641680](https://our.internmc.facebook.com/intern/diff/D34641680)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][refactorization] Refactored qconv_unpack.cpp into an implementation file and higher level call registration and definition file"


Summary:
This refactorization was necessary with the introduction of packed parameters for cudnn.
Specifically, the unpack function for the 3 backends: fbgemm, qnnpack, and cudnn, is called
using dynamic polymorphism, which was previously done in the file (formerly) /quantized/cpu/qconv_unpack.cpp.
This part of the file was moved to the parent directory /quantized/ as it is relevant for both
CPU (fbgemm & qnnpack) & CUDA (cudnn in this case). The remaining content is implementation specific to
CPU, and the file was subsequently renamed to qconv_unpack_impl.cpp

Differential Revision: [D34641680](https://our.internmc.facebook.com/intern/diff/D34641680)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][refactorization] Refactored qconv_unpack.cpp into an implementation file and higher level call registration and definition file"


Summary:
This refactorization was necessary with the introduction of packed parameters for cudnn.
Specifically, the unpack function for the 3 backends: fbgemm, qnnpack, and cudnn, is called
using dynamic polymorphism, which was previously done in the file (formerly) /quantized/cpu/qconv_unpack.cpp.
This part of the file was moved to the parent directory /quantized/ as it is relevant for both
CPU (fbgemm & qnnpack) & CUDA (cudnn in this case). The remaining content is implementation specific to
CPU, and the file was subsequently renamed to qconv_unpack_impl.cpp

Differential Revision: [D34641680](https://our.internmc.facebook.com/intern/diff/D34641680)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][refactorization] Refactored qconv_unpack.cpp into an implementation file and higher level call registration and definition file"


Summary:
This refactorization was necessary with the introduction of packed parameters for cudnn.
Specifically, the unpack function for the 3 backends: fbgemm, qnnpack, and cudnn, is called
using dynamic polymorphism, which was previously done in the file (formerly) /quantized/cpu/qconv_unpack.cpp.
This part of the file was moved to the parent directory /quantized/ as it is relevant for both
CPU (fbgemm & qnnpack) & CUDA (cudnn in this case). The remaining content is implementation specific to
CPU, and the file was subsequently renamed to qconv_unpack_impl.cpp

Differential Revision: [D34641680](https://our.internmc.facebook.com/intern/diff/D34641680)

[ghstack-poisoned]
